docs/protocol/interchain-gas-payment.mdx

```diff
@@ -31,17 +31,32 @@ flowchart LR
 
 Interchain gas payments are facilitated by `InterchainGasPaymaster` (IGP) smart contracts.
 
-These contracts expose the [InterchainGasPaymaster interface](../reference/hooks/interchain-gas.mdx), which allow users to pay the [relayer](./agents/relayer.mdx) with native tokens on the origin chain to cover the costs of delivering a message on the destination chain.
+These contracts expose the [InterchainGasPaymaster interface](../reference/hooks/interchain-gas.mdx), which allow message senders to pay the [relayer](./agents/relayer.mdx) with native tokens on the origin chain to cover the costs of delivering a message on the destination chain.
 
 Each `InterchainGasPaymaster` contract corresponds to exactly one relayer. You can find addresses of the Abacus Works' relayer under [addresses](../reference/contract-addresses.mdx)
 
+### Interchain gas payment calculation
+
+The interchain gas payment is calculated as follows:
+
+$$
+destinationTxCost = destinationGasPrice * gasLimit
+\\[5pt]
+exchangeRate = \frac{originGasTokenPrice}{destinationGasTokenPrice}
+\\[5pt]
+originFee = exchangeRate * destinationTxCost
+$$
+
+The `gasLimit` and the `exchangeRate` are configured by the message sender and the relayer, respectively.
+
 ### Gas limit
 
-The gas limit is the cost of calling `handle` on the destination chain for a given message. This can vary depending on message content and the logic of the handler.
+The gas limit is set based on the cost of calling `handle` on the destination chain for a given message. This can vary depending on message content and the logic of the handler. The message sender is expected to configure the gas limit based on the expected cost of handling the message, a default is used when not configured, see the [interchain gas hook reference](../reference/hooks/interchain-gas.mdx#post-dispatch) for more details.
```

Comment:
```suggestion
The gas limit is set based on the cost of calling `handle` on the destination chain for a given message. This can vary depending on message content and the logic of the handler. The message sender is expected to configure the gas limit based on the expected cost of handling the message, with a default being used when the gas limit is not configured.

See the [interchain gas hook reference](../reference/hooks/interchain-gas.mdx#post-dispatch) for more details.
```